demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py

```diff
@@ -227,7 +226,10 @@ def generate_xql_query(rule: SingleModelingRule, test_data_event_ids: List[str])
 
 
 def validate_expected_values(
```

Comment:
Let's split this function into 2 different atomic logics. 
With your change, it is not `validate_expected_values`... It is doing 2 different checks if some flag is set to true. 
This is very confusing. 
Let's discuss tomorrow my suggestion.